sopel/plugins/handlers.py

```diff
@@ -330,3 +330,51 @@ def reload(self):
         module might not be available through ``sys.path``.
         """
         self._module = self._load()
+
+
+class EntrypointPluginHandler(PyModulePlugin):
+    """Sopel plugin loaded from a ``setuptools`` entry points
+
+    :param entrypoint: a ``setuptools`` entry point object
+
+    This handler loads a Sopel plugin exposed by a ``setuptools`` entry point
+    from the ``sopel.plugins`` group. It expects to be able to load a module
+    object from the entry point, and to work as a :class:`~.PyModulePlugin`
+    from that module.
+
+    To configure a ``sopel.plugins`` entry point, developers must define
+    an entry point either in their ``setup.py`` file or their ``setup.cfg``
+    file::
+
+        setup(
+            name='my_plugin',
+            version='1.0'
+            entry_points={
+                'sopel.plugins': [
+                    'custom = my_plugin.path.to.plugin',
+                ],
+            }
+        )
+
+    And this plugin can be loaded with::
+
+        >>> from pkg_resources import iter_entry_points
+        >>> from sopel.plugins.handlers import EntrypointPluginHandler
+        >>> plugin = [
+        ...     EntrypointPluginHandler(ep)
+        ...     for ep in iter_entry_points('sopel.plugins', 'custom')
+        ... ][0]
+        >>> plugin.load()
+        >>> plugin.name
+        'custom'
+
+    In this example, the plugin ``custom`` is loaded from an entry point.
+    Unlike the :class:`~.PyModulePlugin`, the name is not derived from the
+    actual python module, but from its entry point's name.
+    """
+    def __init__(self, entrypoint):
+        self.entrypoint = entrypoint
```

Comment:
Probably should make the parameter AND the instance variable both `entry_point`, for consistency with Python's documentation, the terminology used in `setup.py`/`setup.cfg`, and the documentation here that refers to "entry points" with that two-word phrase (instead of squashing it into one word).

Reply:
```suggestion
        self.entry_point = entry_point
```